File: skynda_backend/src/java/me/skynda/common/interfaces/services/IVehicleReviewService.java
package me.skynda.common.interfaces.services;

import me.skynda.common.dto.CreateOrUpdateResponseDto;
import me.skynda.common.dto.DeleteResponseDto;
import me.skynda.vehicle.dto.VehicleReviewAdminDto;
import org.springframework.validation.BindingResult;

import java.io.Serializable;
import java.util.List;


public interface IVehicleReviewService {
    List<VehicleReviewAdminDto> getAll();

    VehicleReviewAdminDto getSingleBy(Integer id);

    CreateOrUpdateResponseDto createOrUpdate(VehicleReviewAdminDto dto, BindingResult bindingResult);

    DeleteResponseDto delete(Integer id);

    List<VehicleReviewAdminDto> getAllBy(Serializable vehicleId);

}

File: skynda_dotnet_backend/Triven.Domain/Services/IVehicleReviewService.cs
using System.Collections.Generic;
using Triven.Domain.Results;
using Triven.Domain.ViewModels.Vehicle;

namespace Triven.Domain.Services
{
    public interface IVehicleReviewService
    {
        ServiceResult<IEnumerable<VehicleReviewViewModel>> GetAll();
        ServiceResult<VehicleReviewViewModel> Get(int id);
        ServiceResult<VehicleReviewViewModel> Create(VehicleReviewViewModel dto);
        ServiceResult<VehicleReviewViewModel> Update(int id, VehicleReviewViewModel dto);
        ServiceResult<bool> Delete(int id);
        ServiceResult<IList<VehicleReviewViewModel>> GetAllBy(int vehicleId);
    }
}